Refactoring the way we generate volume/volumeMounts in build/sign flows.

ybettan opened the pull request:
To be used in the build flow as well in the build-sign flow.

---

/assign @yevgeny-shnaidman

yevgeny-shnaidman commented:
1) i like it that the volume and volume mount are explicit :)
2) WDYT about moving the volumes/volumesmount to a common_volumes.go file? the common.go becomes a large file
3) why do we need to change sign package and delete the utils in this PR? It is a little bit confusing since we are already not using sign package. We can leave till we the PR where we delete build and sign packages

ybettan commented:
> i like it that the volume and volume mount are explicit :)

I am happy you liked it :)

> WDYT about moving the volumes/volumesmount to a common_volumes.go file? the common.go becomes a large file

I moved it to a new `volumes.go` file in the same directory.

> why do we need to change sign package and delete the utils in this PR? It is a little bit confusing since we are already not using sign package. We can leave till we the PR where we delete build and sign packages

Because at first I was using methods (instead of explicit volume/volumeMount list) so I have created new methods in the buildsign/pod package and removed them from utils.

mresvanis commented:
/lgtm
